### DISCUSSION TITLE
What is essential and nice-to-have for a Dagger UI?

### DISCUSSION THREAD

**Discussion by jpadams** [dagger/dagger#3800] [General]
One element mentioned is a DAG viewer: https://github.com/dagger/dagger/issues/2144

What else?

**Comment by marcosnils** [comment]
Logs scoped to each DAG node

**Comment by benjaminstrasser** [comment]
Jump to code/Open Editor like [Cypress IDE Integration](https://docs.cypress.io/guides/tooling/IDE-integration#File-Opener-Preference)

**Comment by nipuna-perera** [comment]
Pipeline metrics would be awesome. For example, time taken to run each
section. Some kind of cached historical metrics would also be nice to drive
trends and to calculate DORA metrics.

**Comment by beingminimal** [comment]
@jpadams how we can use https://github.com/excalidraw/excalidraw to visualise dagger pipelines live which will help to debug easily at monorepo use cases
